### PR TITLE
re-analyze quoted subscript expression if necessary

### PR DESCRIPTION
**Pull request by chaudum**
this is later used for ODBC compatibility

### CODE REVIEW

**Comment by mikethebeer** [comment]
ack

File: sql/src/test/java/io/crate/analyze/expressions/ExpressionAnalyzerTest.java [hunk 1]
```diff
@@ -108,6 +109,22 @@ public void testUnsupportedExpressionCurrentDate() throws Exception {
...
+
+        Field field1 = (Field) expressionAnalyzer.convert(SqlParser.createExpression("obj['x']"), expressionAnalysisContext);
+        Field field2 = (Field) expressionAnalyzer.convert(SqlParser.createExpression("\"obj['x']\""), expressionAnalysisContext);
+        assertEquals(field1, field2);
+
+        assertFalse(ExpressionAnalyzer.isQuotedSubscript(new QualifiedNameReference(new QualifiedName("obj"))));
+        assertFalse(ExpressionAnalyzer.isQuotedSubscript(new QualifiedNameReference(new QualifiedName("obj.x"))));
+        assertFalse(ExpressionAnalyzer.isQuotedSubscript(new QualifiedNameReference(new QualifiedName("obj[x][y]"))));
+        assertTrue(ExpressionAnalyzer.isQuotedSubscript(new QualifiedNameReference(new QualifiedName("obj['x']"))));
+        assertTrue(ExpressionAnalyzer.isQuotedSubscript(new QualifiedNameReference(new QualifiedName("obj['x']['y']"))));
```
Comment (seut) [inline sql/src/test/java/io/crate/analyze/expressions/ExpressionAnalyzerTest.java]: pls add a test validating a subscript expr where quoting is really necessary like e.g. "myObj['x']"

**Comment by seut** [comment]
pls use pre-compiled patterns, otherwise ack

**Comment by seut** [comment]
ack

**Pull request merged**